### DISCUSSION TITLE
Code splitting for lazy loading igniteui-angular

### DISCUSSION THREAD

**Discussion by lalo-mx** [IgniteUI/igniteui-angular#13856] [Q&A] [answered]
Currently the library is published as a single entrypoint. Using the library with Angular produces a bundle with igniteui-angular in main. 
It would be nice to be able to have multiple chunks for the lazy modules.

![image](https://github.com/IgniteUI/igniteui-angular/assets/7459881/f8e91276-2157-4016-8208-e7f92e431380)

![image](https://github.com/IgniteUI/igniteui-angular/assets/7459881/9b9c4aea-99ce-403e-a84d-75320c1deaa7)

Any feeback if this is possible or desired?

https://angular.io/guide/angular-package-format#entrypoints-and-code-splitting
```
The general rule for APF packages is to use entrypoints for the smallest sets of logically connected code possible. 
For example, the Angular Material package publishes each logical component or set of components 
as a separate entrypoint - one for Button, one for Tabs, etc. This allows each Material component
 to be lazily loaded separately, if desired.
```

**Comment by kdinev** [comment, marked as answer]
@lalo-mx We have already considered this, and the conclusions were the following:

1. The change would be large/potentially breaking for mature projects already dependent on `igniteui-angular` (migrations can mitigate that)
2. The change would be large for the repository itself, introducing risk.

We already moved animations as a separate entry point: `igniteui-angular/animations` and for the past couple of releases we've focused a lot of efforts on tree-shaking improvements in the library. You can check #13562.
We will do some further research into how beneficial multiple entry points would be for large projects. If we find that the change is worth it, then we will start introducing it no earlier than `18.0.0`.
